eth-contracts/test/governance.test.js

```diff
@@ -610,99 +619,226 @@ contract('Governance.sol', async (accounts) => {
-      assert.equal(_lib.toStr(proposal.targetContractRegistryKey), _lib.toStr(targetContractRegistryKey), 'Expected same proposal.targetContractRegistryKey')
-      assert.equal(proposal.targetContractAddress, targetContractAddress, 'Expected same proposal.targetContractAddress')
-      assert.equal(fromBN(proposal.callValue), callValue, 'Expected same proposal.callValue')
-      assert.equal(proposal.signature, signature, 'Expected same proposal.signature')
-      assert.equal(proposal.callData, callData, 'Expected same proposal.callData')
-      assert.equal(proposal.outcome, outcome, 'Expected same outcome')
-      assert.equal(parseInt(proposal.voteMagnitudeYes), defaultStakeAmount, 'Expected same voteMagnitudeYes')
-      assert.equal(parseInt(proposal.voteMagnitudeNo), 0, 'Expected same voteMagnitudeNo')
-      assert.equal(parseInt(proposal.numVotes), 1, 'Expected same numVotes')
-
-      // Confirm that contract was upgraded by ensuring staking.newFunction() call succeeds
-      const stakingCopy2 = await StakingUpgraded.at(staking.address)
-      const newFnResp = await stakingCopy2.newFunction.call({ from: proxyDeployerAddress })
-      assert.equal(newFnResp, 5)
-
-      // Confirm that proxy contract's implementation address has upgraded
+      const guardianExecTx = _lib.parseTx(guardianExecTxReceipt)
+      assert.equal(guardianExecTx.event.name, 'GuardianTransactionExecuted', 'event.name')
+      assert.equal(guardianExecTx.event.args.targetContractAddress, targetContractAddress, 'event.args.targetContractAddress')
+      assert.isTrue(guardianExecTx.event.args.callValue.eq(callValue), 'event.args.callValue')
+      // assert.equal(guardianExecTx.event.args.signature, web3.utils.utf8ToHex(signature), 'event.args.signature')
+      // assert.equal(guardianExecTx.event.args.callData, callData, 'event.args.callData')
+      assert.equal(guardianExecTx.event.args.success, true, 'event.args.success')
+      assert.equal(guardianExecTx.event.args.returnData, returnData, 'event.args.returnData')
+
+      // Confirm Slash action succeeded by checking new Stake + Token values
+      const finalStakeAcct2 = await staking.totalStakedFor(targetAddress)
+      assert.isTrue(
+        finalStakeAcct2.eq(defaultStakeAmount.sub(web3.utils.toBN(slashAmount)))
+      )
+      assert.isTrue(
+        (initialTotalStake.sub(slashAmount)).eq(await staking.totalStaked()),
+        'Expected same total stake amount'
+      )
       assert.equal(
-        await stakingProxy.implementation.call({ from: proxyAdminAddress }),
-        stakingUpgraded0.address,
-        'Expected updated proxy implementation address'
+        await token.totalSupply(),
+        initialTokenSupply - slashAmount,
+        "Expected same token total supply"
       )
     })
+
+    it.skip('Fail to slash too large amount', async () => {
+
+    })
+
+    it.skip('Upgrade contract', async () => {
```

Comment:
i think this one has been covered above?

Reply:
nah this is separate. i want to add a test for a direct guardianExzecuteTransaction for upgrade, rather than submitting a proposal to upgrade.